synapseclient/client.py

```diff
@@ -1040,17 +1058,14 @@ def store(self, obj, *, createOrUpdate=True, forceVersion=True, versionLabel=Non
                                                        requestedObjects={'includeEntity': True,
                                                                          'includeAnnotations': True})
 
-                    # Need some fields from the existing entity: id, etag, and version info.
-                    existing_entity = bundle['entity']
+                    properties, annotations = self._merge_update_with_bundle(
+                        bundle,
+                        properties,
+                        annotations,
+                    )
 
-                    # Update the conflicting Entity
-                    existing_entity.update(properties)
-                    properties = self._updateEntity(existing_entity, forceVersion, versionLabel)
+                    properties = self._updateEntity(properties, forceVersion, versionLabel)
 
-                    # Merge new annotations with existing annotations
-                    existing_annos = from_synapse_annotations(bundle['annotations'])
```

Comment:
In this case since an ID didn't previously exist, we don't treat missing annotations as deletions since they might have just created an new Entity object w/ no annotations initialized.